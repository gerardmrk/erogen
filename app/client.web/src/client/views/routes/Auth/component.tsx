import * as React from "react";
import styles from "./component.styles.scss";
import HeadTags from "@client/views/components/HeadTags";
import Container from "@client/views/components/ui-elements/Container";
import { LocalProps } from ".";

export type Props = LocalProps;

export type State = {};

export class Auth extends React.PureComponent<Props, State> {
  public static i18nNamespace = "route_Auth";

  public render() {
    return (
      <React.Fragment>
        <HeadTags
          path={this.props.path}
          title={this.props.t("title")}
          description={this.props.t("description")}
          metaType={this.props.metaType}
          metaImgPath={this.props.metaImgPath}
          metaImgAlt={this.props.metaImgAlt}
          metaTwitterCardType={this.props.metaTwitterCardType}
        />

        <div className={styles.main}>
          <Container>
            <h1>{"Auth"}</h1>
          </Container>
        </div>
      </React.Fragment>
    );
  }
}
