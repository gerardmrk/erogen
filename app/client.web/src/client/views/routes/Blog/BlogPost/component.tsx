import * as React from "react";
import styles from "./component.styles.scss";

type Props = {};

type State = {};

export class BlogPost extends React.PureComponent<Props, State> {
  public static readonly chunkName = "Blog-BlogPost";
  public static readonly i18nNamespace = "";

  public render() {
    return (
      <div className={styles.main}>
        <h1>{"BlogPost"}</h1>
      </div>
    );
  }
}

export default BlogPost;
