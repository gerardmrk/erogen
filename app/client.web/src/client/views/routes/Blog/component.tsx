import * as React from "react";
import styles from "./component.styles.scss";
import { Switch } from "react-router-dom";
import { LocalProps } from ".";
import EnhancedRoute from "@client/views/components/EnhancedRoute";
import Container from "@client/views/components/ui.elements/Container";
import HeadTags from "@client/views/components/HeadTags";

export type Props = LocalProps;

export type State = {};

export class Blog extends React.PureComponent<Props, State> {
  public static i18nNamespace = "route_Blog";

  public render() {
    return (
      <React.Fragment>
        <HeadTags
          path={this.props.path}
          title={this.props.t("title")}
          description={this.props.t("description")}
          metaType={this.props.metaType}
          metaImgPath={this.props.metaImgPath}
          metaImgAlt={this.props.metaImgAlt}
          metaTwitterCardType={this.props.metaTwitterCardType}
        />

        <div className={styles.main}>
          <Container>
            <h1>{"Blog"}</h1>

            <Switch>
              {this.props.routes.map((r, i) => (
                <EnhancedRoute key={i} {...r} />
              ))}
            </Switch>
          </Container>
        </div>
      </React.Fragment>
    );
  }
}
