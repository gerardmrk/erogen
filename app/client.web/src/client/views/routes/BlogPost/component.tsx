import * as React from "react";
import styles from "./component.styles.scss";
import { LocalProps } from ".";
import HeadTags from "@client/views/components/HeadTags";

type Props = LocalProps;

type State = {};

export class BlogPost extends React.PureComponent<Props, State> {
  public static i18nNamespace = "route_BlogPost";

  public render() {
    return (
      <React.Fragment>
        <HeadTags
          path={this.props.path}
          title={this.props.t("title")}
          description={this.props.t("description")}
          metaType={this.props.metaType}
          metaImgPath={this.props.metaImgPath}
          metaImgAlt={this.props.metaImgAlt}
          metaTwitterCardType={this.props.metaTwitterCardType}
        />

        <div className={styles.main}>
          <h1>{"BlogPost"}</h1>
        </div>
      </React.Fragment>
    );
  }
}
