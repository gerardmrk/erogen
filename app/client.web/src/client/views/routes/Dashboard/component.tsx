import * as React from "react";
import styles from "./component.styles.scss";

type Props = {};

type State = {};

export class Dashboard extends React.PureComponent<Props, State> {
  public render() {
    return (
      <div className={styles.main}>
        <h1>{"Dashboard"}</h1>
      </div>
    );
  }
}
