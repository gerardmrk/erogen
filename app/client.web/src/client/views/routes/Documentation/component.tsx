import * as React from "react";
import { Heading } from "evergreen-ui/esm/typography";
import styles from "./styles.scss";
import Container from "@client/views/components/Container";

type Props = {};

type State = {};

export class Documentation extends React.PureComponent<Props, State> {
  public render() {
    return (
      <div className={styles.main}>
        <Container>
          <Heading size={700}>{"Documentation"}</Heading>
        </Container>
      </div>
    );
  }
}

export default Documentation;
