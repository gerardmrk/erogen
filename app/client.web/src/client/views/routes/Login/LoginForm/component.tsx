import * as React from "react";
import Form from "@client/views/components/ui.collections/Form";
import styles from "./component.styles.scss";

type Props = {};

type State = {};

export class LoginForm extends React.PureComponent<Props, State> {
  public render() {
    return (
      <Form className={styles.main}>
        <Form.Group widths={"equal"}>
          <Form.Input type={"text"} autoComplete={"username"} fluid={true} />
        </Form.Group>

        <Form.Group widths={"equal"}>
          <Form.Input
            type={"password"}
            autoComplete={"current-password"}
            fluid={true}
          />
        </Form.Group>
      </Form>
    );
  }
}
