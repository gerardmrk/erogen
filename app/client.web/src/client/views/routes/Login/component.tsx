import * as React from "react";
import styles from "./component.styles.scss";
import AuthRoutesWrapper from "@client/views/core/AuthRoutesWrapper";
import LoginForm from "./LoginForm";

type Props = {};

type State = {};

export class Login extends React.PureComponent<Props, State> {
  public render() {
    return (
      <AuthRoutesWrapper title={"Login"}>
        <div className={styles.main}>
          <LoginForm />
        </div>
      </AuthRoutesWrapper>
    );
  }
}
