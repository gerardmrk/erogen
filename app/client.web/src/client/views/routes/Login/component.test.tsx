import * as React from "react";
import { shallow } from "enzyme";
import { Login } from "./component";

describe("<Login/>", () => {
  it("renders ok", () => {
    const wrapper = shallow(<Login />);
    expect(wrapper).toExist();
  });

  it("has static properties required for a route", () => {
    expect(Login.chunkName).toBeDefined();
    expect(typeof Login.chunkName).toEqual("string");

    expect(Login.i18nNamespace).toBeDefined();
    expect(typeof Login.i18nNamespace).toEqual("string");
  });
});
