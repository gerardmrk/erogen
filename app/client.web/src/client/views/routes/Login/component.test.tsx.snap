// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> snapshot 1`] = `
<div
  className="main"
>
  <div
    className="container"
  >
    <div
      className="ui fluid card"
      onClick={[Function]}
    >
      <div
        className="content header"
      >
        <div
          className="header"
        >
          <h1>
            form-heading
          </h1>
        </div>
      </div>
      <div
        className="content"
      >
        <div
          className="content"
        >
          <div
            className="main"
          >
            <form
              className="ui form main"
              onSubmit={[Function]}
            >
              <div
                className="field"
                widths="equal"
              >
                <label>
                  <span>
                    form-label-username
                  </span>
                  <div
                    className="ui fluid input"
                  >
                    <input
                      autoComplete="username"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                  </div>
                </label>
              </div>
              <div
                className="field"
                widths="equal"
              >
                <label>
                  <span>
                    form-label-password
                  </span>
                  <div
                    className="ui fluid input"
                  >
                    <input
                      autoComplete="current-password"
                      onChange={[Function]}
                      type="password"
                      value=""
                    />
                  </div>
                </label>
              </div>
              <div
                className="field"
              >
                <div
                  className="ui checkbox"
                  onChange={[Function]}
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  onMouseUp={[Function]}
                >
                  <input
                    checked={false}
                    className="hidden"
                    readOnly={true}
                    tabIndex={0}
                    type="checkbox"
                  />
                  <label>
                    form-label-remember
                  </label>
                </div>
              </div>
              <button
                className="ui button"
                onClick={[Function]}
                type="submit"
              >
                form-heading
              </button>
            </form>
          </div>
          <nav
            className="main"
          >
            nav1 nav2 nav3
          </nav>
        </div>
      </div>
    </div>
  </div>
</div>
`;
